File: app/src/main/java/com/rendersoncs/reportform/listener/OnInteractionListener.java
package com.rendersoncs.reportform.listener;

import java.io.File;

public interface OnInteractionListener {

    /* Click for List Report*/
    void onListClick(int reportId);

    /* Click open PDF*/
    void onOpenPdf(int reportId);

    /* Click for Delete Report*/
    void onDeleteClick(int id);

    /* Click for Share Report*/
    void onShareReport(int reportId);

    /* Click for Edit Report*/
    void onEditReport(int id);
}

File: app/src/main/java/com/rendersoncs/reportform/view/adapter/listener/OnInteractionListener.kt
package com.rendersoncs.reportform.view.adapter.listener

interface OnInteractionListener {
    /* Click for List Report*/
    fun onListClick(id: Int)

    /* Click open PDF*/
    fun onOpenPdf(id: Int)

    /* Click for Delete Report*/
    fun onDeleteClick(id: Int)

    /* Click for Share Report*/
    fun onShareReport(id: Int)

    /* Click for Edit Report*/
    fun onEditReport(id: Int)
}